Col/looper

aclong28 opened the pull request:
feature: separate each class for main_window and editor_window

lehoang318 committed:
Update Main Window

- Remove Toolbar, Export & Save Actions
- Rename Merge Actions

lehoang318 committed:
Create Editor Dialog

lehoang318 committed:
Optimize display performance for 16-bit images

lehoang318 committed:
Update UI

- Remove unused actions
- Link Editor Window to Main Window
- Create resource file: icons.qrc

lehoang318 committed:
Create execution script: run.sh

lehoang318 committed:
Update run.sh to detect environment

lehoang318 committed:
Update designs

- Correct Image, Images classes' structures
- Add new usecase: Execute

aclong28 committed:
feature: separate each class for main_window and editor_window

lehoang318 committed:
Support Dependencies Check

lehoang318 committed:
Update comments & guidelines

lehoang318 committed:
Add executable permission to run.sh

aclong28 committed:
feature: create new commonfunction class to display image

aclong28 committed:
feature: remove display method in editor_window and main_window

lehoang318 committed:
Implement Support Utilities

- Support Execution
  + Check dependencies
  + Check UI, Resource files, and re-generate if necessary

- Support Cleaning up Workspace

aclong28 committed:
Merge pull request #2 from lehoang318/execution_support_dev

Implement Support Utilities

lehoang318 commented:
- [ui] Rename editor.ui -> ui_editor_window.ui
- [ui] Rename main_window.ui -> ui_main_window.ui
- [wrapper.py] Change display() to static with two arguments: labelCanvas & image
- [editor_window.py] class EditorWindow inherits from class CommonFunctions
- [editor_window.py] Remove EditorWindow.actionEditorClickEvt()
- [editor_window.py] __init__(): editor & gui are dupplicated
- [main_window.py] class MainWindow inherits from class CommonFunctions
- [main_window.py] actionCurrentItemChangedEvt() should not return the selected image
- [main_window.py] actionEditClickEvt() should get the select image directly by index
  self.gui.listWidgetImgList.currentRow()

aclong28 committed:
feature: fix as comment

aclong28 commented:
> * [ui] Rename editor.ui -> ui_editor_window.ui
--> extension is ".ui" so not needed
> * [ui] Rename main_window.ui -> ui_main_window.ui
--> extension is ".ui" so not needed
> * [wrapper.py] Change display() to static with two arguments: labelCanvas & image
--> done
> * [editor_window.py] class EditorWindow inherits from class CommonFunctions
--> These are different classes cannot be inherit
> * [editor_window.py] Remove EditorWindow.actionEditorClickEvt()
--> done
> * [editor_window.py] **init**(): editor & gui are dupplicated
--> done
> * [main_window.py] class MainWindow inherits from class CommonFunctions
--> These are different classes cannot be inherit
> * [main_window.py] actionCurrentItemChangedEvt() should not return the selected image
--> why ?
> * [main_window.py] actionEditClickEvt() should get the select image directly by index
>   self.gui.listWidgetImgList.currentRow()
--> why ?

aclong28 committed:
feature: rename ui files

lehoang318 committed:
Update the scripts:

- Update pyui_compiler.py to be compatible with Linux environment
- Add GNU General Public License to pyui_compiler.py

lehoang318 committed:
Implement Image Rotation:

- Change constructor of Image class
- Change structure of MainWindow, EditorWindow classes
  + Parent: QtWidgets.QMainWindow
- Change title of EditorWindow
- Add new attribute to EditorWindow: editingImage
- Add new method to MainWindow: terminateEditorWindow()

lehoang318 committed:
Implement Image Rotation

- Add *.bak to .gitignore
- Update design files
- Create State class to store information about editing status
- Separate LOG() from main module
- Create back-link from EditorWindow to MainWindow
- Create Rotation Dialog

aclong28 committed:
feature: create resize image util

lehoang318 committed:
Implement Image Rotation

- Rename ui/ui_slider_dialog.ui -> ui/ui_rotation_dialog.ui
- Add Reset Action to EditorWindow
- Create Reset icon

lehoang318 committed:
Implement Image Rotation

- Remove the second constructor in State class
- Create Action Events for EditorWindow

aclong28 committed:
resize_image: fix cannot open save window and open filedialogue to save

lehoang318 committed:
Implement Image Rotation

- Update ui/ui_rotation_dialog.ui
- Support basic rotation
 + Arbitrary angles (slider adjustment)
 + ROTATE_90_CLOCKWISE ('Right' button)
 + ROTATE_90_COUNTERCLOCKWISE ('Left' button)

aclong28 committed:
resize-image: add option to resize with percentage

lehoang318 committed:
Implement Image Rotation

- Support Image Flipping (Horizontal & Vertical)

lehoang318 committed:
Add rotation_dialog.py

lehoang318 committed:
Fix bug related to Windows/Dialogs closing:

- Finalize sub windows/dialogs when parent widget is closed
- Improve LOG() function with traceability information

lehoang318 committed:
Create Wrapper classes to support sub-widgets management

aclong28 committed:
Merge pull request #3 from lehoang318/image_rotation_dev

Image rotation dev

aclong28 committed:
Merge branch 'col/looper' of github.com:lehoang318/img-pro into feature/resize-image

aclong28 committed:
resize-image: use alternate way to call qt object

lehoang318 committed:
Merge pull request #4 from lehoang318/feature/resize-image

Feature/resize image
